pkg/agent/cniserver/server_windows_test.go

```diff
@@ -0,0 +1,1038 @@
+
+	for _, tc := range []struct {
+		name           string
+		podName        string
+		containerID    string
+		netns          string
+		ipamDel        bool
+		ipamError      error
+		endpointExists bool
+		disconnectOVS  bool
+		ifaceExists    bool
+		response       *cnipb.CniCmdResponse
+	}{
+		{
+			name:           "docker-infra-success",
+			podName:        "pod0",
+			containerID:    containerID,
+			netns:          "none",
+			ipamDel:        true,
+			disconnectOVS:  true,
+			endpointExists: true,
+			ifaceExists:    true,
+		}, {
+			name:        "interface-not-exist",
+			podName:     "pod1",
+			containerID: containerID,
+			netns:       "none",
+			ipamDel:     true,
+		}, {
+			name:        "ipam-delete-failure",
+			podName:     "pod2",
+			containerID: containerID,
+			netns:       "none",
+			ipamDel:     true,
+			ipamError:   fmt.Errorf("unable to delete IP"),
+			response: &cnipb.CniCmdResponse{
+				Error: &cnipb.Error{
+					Code:    cnipb.ErrorCode_IPAM_FAILURE,
+					Message: "unable to delete IP",
+				},
+			},
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			isDocker := isDockerContainer(tc.netns)
+			requestMsg, ovsPortName := prepareSetup(t, ipamType, tc.podName, tc.containerID, tc.containerID, tc.netns, nil)
+			hnsEndpoint := getHnsEndpoint(generateUUID(t), ovsPortName)
+			var existingHnsEndpoints []hcsshim.HNSEndpoint
+			if tc.endpointExists {
+				existingHnsEndpoints = append(existingHnsEndpoints, *hnsEndpoint)
+			}
+			testUtil := newHnsTestUtil(hnsEndpoint.Id, existingHnsEndpoints, isDocker, true, nil, nil)
+			testUtil.setFunctions()
+			waiter := newAsyncWaiter(tc.podName, tc.containerID)
+			server := mockCNIServer(t, controller, waiter.notifier)
+			ovsPortID := generateUUID(t)
+			if tc.ifaceExists {
+				containerIface := interfacestore.NewContainerInterface(ovsPortName, tc.containerID, tc.podName, testPodNamespace, containerMAC, []net.IP{net.ParseIP("10.1.2.100")}, 0)
+				containerIface.OVSPortConfig = &interfacestore.OVSPortConfig{
+					OFPort:   100,
+					PortUUID: ovsPortID,
+				}
+				ifaceStore.AddInterface(containerIface)
+			}
+			if tc.ipamDel {
+				ipamMock.EXPECT().Del(gomock.Any(), gomock.Any(), gomock.Any()).Return(true, tc.ipamError).Times(1)
+			}
+			if tc.disconnectOVS {
+				mockOVSBridgeClient.EXPECT().DeletePort(ovsPortID).Return(nil).Times(1)
+				mockOFClient.EXPECT().UninstallPodFlows(ovsPortName).Return(nil).Times(1)
+				routeMock.EXPECT().DeleteLocalAntreaFlexibleIPAMPodRule(gomock.Any()).Return(nil).Times(1)
+			}
+			resp, err := server.CmdDel(ctx, requestMsg)
+			assert.NoError(t, err)
+			if tc.response != nil {
+				assert.Equal(t, tc.response, resp)
+			} else {
+				assert.Equal(t, emptyResponse, resp)
+			}
+			_, exists := ifaceStore.GetContainerInterface(tc.containerID)
+			assert.False(t, exists)
+			if tc.endpointExists {
+				_, exists = server.podConfigurator.ifConfigurator.(*ifConfigurator).getEndpoint(ovsPortName)
```

Comment:
Should we check it exists before we call `server.CmdDel(ctx, requestMsg)`?

Reply:
Add a call to explicitly add the hnsEndpoint into cache before CmdDel.